engine/crates/engine-v2/src/sources/graphql/federation.rs

```diff
@@ -101,12 +101,15 @@ pub(crate) struct FederationEntityExecutor<'ctx, R: Runtime> {
 impl<'ctx, R: Runtime> FederationEntityExecutor<'ctx, R> {
     #[tracing::instrument(skip_all)]
     pub async fn execute(self, mut response_part: ResponsePart) -> ExecutionResult<ResponsePart> {
-        let subgraph_gql_request_span = SubgraphRequestSpan::new(self.subgraph.name())
-            .with_url(self.subgraph.url())
-            .with_operation_type(OperationType::Query.as_ref())
-            // The query string contains no input values, only variables. So it's safe to log.
-            .with_document(&self.operation.query)
-            .into_span();
+        let span = SubgraphRequestSpan {
+            name: self.subgraph.name(),
+            operation_type: OperationType::Query.as_str(),
+            // The generated query does not contain any data, everything are in the variables, so
+            // it's safe to use.
+            sanitized_query: &self.operation.query,
```

Comment:
Is this still sanitized though, should we consider a different name?

Reply:
yes it is, the query generated for the subgraph has no query variables except enums. Everything else are passed on as variables

Reply:
It's still a bit different, no? The sanitized query e.g. changes all strings to empty strings, all numbers to 0 etc.

Reply:
that's the normalized query you're talking about. Hence why I chose "sanitized" here. I'm not super convinced normalized query is that much useful to be honest, but it's also sanitized so well works for now.

Reply:
the only goal here is to prevent any sensitive data leaks in the logs. Query normalization is used for analytics currently, which is different and where I have my doubts. But as said it's sanitized so also works for the logs.

Reply:
Fair. Sashay away!